### ISSUE TITLE
Support gitlab-style inline math $`a=b`$

### ISSUE THREAD

**Issue by kalvdans** [CrossNox/m2r2#20]
Since https://github.com/miyakogi/m2r/commit/eeb618b9e5a3c59b90e4c5ac92df0703c57ac76a m2r supports inline math with dollar signs inside backticks, but [gitlab](https://docs.gitlab.com/ee/user/markdown.html#math) uses backticks inside dollar signs.

We would like to render the same document with gitlab and with sphinx, and so would like m2r2 to support the gitlab way.

**Comment by kalvdans** [comment]
Since 15.5 gitlab supports inline math surrounded by only dollar signs only: https://gitlab.com/gitlab-org/gitlab/-/issues/21757

Their regular expression to avoid treating currencies as math is [here](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/94111/diffs#257c616eda9b5a63fb6355844cee1e9ef85da559_18_23)

**Comment by alexisthual** [comment]
I agree with @kalvdans and think that `m2r2` should consider that inline math will be using `$...$` instead of `` `$...$` ``.
I guess all it takes is to change this line:
https://github.com/CrossNox/m2r2/blob/0408d7acea843485d9ff42ee08a105a79f045493/m2r2.py#L122
to
```python
inline_math = re.compile(r"\$(.*?)\$")
```

What do you think @CrossNox ? :blush: 
Thanks for the great package anyways!

**Comment by milannnnn** [comment]
Any news regarding this issue?

**Comment by kalvdans** [comment]
Pull request up at https://github.com/CrossNox/m2r2/pull/62; please try it out @milannnnn and @alexisthual!

We hit https://gitlab.com/gitlab-org/gitlab/-/issues/389949 preventing some expressions to render correctly in gitlab; the generated mathjax with m2r2 + sphinx looks ok though.
